subsys/bluetooth/services/fast_pair/fp_gatt_service.c

```diff
@@ -432,24 +704,50 @@ static ssize_t write_account_key(struct bt_conn *conn,
 	return res;
 }
 
-BT_GATT_SERVICE_DEFINE(fast_pair_svc,
-BT_GATT_PRIMARY_SERVICE(BT_UUID_FAST_PAIR),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_MODEL_ID,
-		BT_GATT_CHRC_READ,
-		BT_GATT_PERM_READ,
-		read_model_id, NULL, NULL),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_KEY_BASED_PAIRING,
-		BT_GATT_CHRC_WRITE | BT_GATT_CHRC_NOTIFY,
-		BT_GATT_PERM_WRITE,
-		NULL, write_key_based_pairing, NULL),
-	BT_GATT_CCC(NULL, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_PASSKEY,
-		BT_GATT_CHRC_WRITE | BT_GATT_CHRC_NOTIFY,
-		BT_GATT_PERM_WRITE,
-		NULL, write_passkey, NULL),
-	BT_GATT_CCC(NULL, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
-	BT_GATT_CHARACTERISTIC(BT_UUID_FAST_PAIR_ACCOUNT_KEY,
-		BT_GATT_CHRC_WRITE,
-		BT_GATT_PERM_WRITE,
-		NULL, write_account_key, NULL),
-);
+static ssize_t write_additional_data(struct bt_conn *conn,
+				     const struct bt_gatt_attr *attr,
+				     const void *buf,
+				     uint16_t len, uint16_t offset, uint8_t flags)
+{
+	/* The only expected Additional Data write is with Personalized Name. fp_keys module will
+	 * return an error if it receives Personalized Name store request in inappropriate state.
+	 */
+	size_t data_len = len - FP_CRYPTO_ADDITIONAL_DATA_HEADER_LEN;
+	uint8_t decoded_data[data_len + sizeof(char)];
+	int err = 0;
+	ssize_t res = len;
+
+	if (offset != 0) {
```

Comment:
What about `len` parameters checking? Isn't needed here?

Reply:
From Fast Pair perspective it can be of any length. From application perspective we are not quite sure how long can the message in callback be. I will create a follow-up task to investigate it further.

Reply:
https://projecttools.nordicsemi.no/jira/browse/NCSDK-17934